influxql/engine.go

```diff
@@ -663,6 +665,31 @@ func ReduceMean(key Key, values []interface{}, e *Emitter) {
 	e.Emit(key, out.Sum/float64(out.Count))
 }
 
+// MapMin collects the values to pass to the reducer
+func MapMin(itr Iterator, e *Emitter, tmin int64) {
+	var values []float64
+
+	for k, v := itr.Next(); k != 0; k, v = itr.Next() {
+		values = append(values, v.(float64))
+	}
+	e.Emit(Key{tmin, itr.Tags()}, values)
+}
+
+// ReduceMin computes the min of value.
+func ReduceMin(key Key, values []interface{}, e *Emitter) {
+	var min float64
+	for i, value := range values {
+		vals := value.([]float64)
+		for j, v := range vals {
+			if i == 0 && j == 0 {
```

Comment:
I see. It's a little obscure for the next guy. I would have simply created a bool flag at the start.

`firstValue := true'

and then:

`if firstValue min = v; firstValue = false`

In exchange, you could then ignore the indexes. This is just a style thing though.